src/compas_pb/conversions.py

```diff
@@ -508,3 +528,553 @@ def arc_from_pb(proto_data: arc_pb2.ArcData) -> Arc:
+    """
+    proto_data = transformation_pb2.TransformationData()
+    proto_data.guid = str(transformation.guid)
+    proto_data.name = transformation.name
+
+    # Flatten 4x4 matrix to list of 16 floats
+    matrix = transformation.matrix
+    for row in matrix:
+        for value in row:
+            proto_data.matrix.append(value)
+
+    return proto_data
+
+
+@pb_deserializer(transformation_pb2.TransformationData)
+def transformation_from_pb(proto_data: transformation_pb2.TransformationData) -> Transformation:
+    """
+    Convert a protobuf message to COMPAS Transformation.
+
+    Parameters
+    ----------
+    proto_data : transformation_pb2.TransformationData
+        The protobuf message representing a Transformation.
+
+    Returns
+    -------
+    Transformation
+        The deserialized COMPAS Transformation object.
+    """
+    # Convert flat list of 16 floats back to 4x4 matrix
+    matrix_flat = list(proto_data.matrix)
+    matrix = []
+    for i in range(4):
+        row = []
+        for j in range(4):
+            row.append(matrix_flat[i * 4 + j])
+        matrix.append(row)
+
+    result = Transformation.from_matrix(matrix)
+    result.name = proto_data.name
+    result._guid = proto_data.guid
+    return result
+
+
+# =============================================================================
+# Translation
+# =============================================================================
+
+
+@pb_serializer(Translation)
+def translation_to_pb(translation: Translation) -> translation_pb2.TranslationData:
+    """
+    Convert a COMPAS Translation to protobuf message.
+
+    Parameters
+    ----------
+    translation : Translation
+        The COMPAS Translation object to serialize.
+
+    Returns
+    -------
+    translation_pb2.TranslationData
+        The protobuf message representing the Translation.
+    """
+    proto_data = translation_pb2.TranslationData()
+    proto_data.guid = str(translation.guid)
+    proto_data.name = translation.name
+
+    vector = vector_to_pb(translation.translation_vector)
+    proto_data.translation_vector.CopyFrom(vector)
+
+    return proto_data
+
+
+@pb_deserializer(translation_pb2.TranslationData)
+def translation_from_pb(proto_data: translation_pb2.TranslationData) -> Translation:
+    """
+    Convert a protobuf message to COMPAS Translation.
+
+    Parameters
+    ----------
+    proto_data : translation_pb2.TranslationData
+        The protobuf message representing a Translation.
+
+    Returns
+    -------
+    Translation
+        The deserialized COMPAS Translation object.
+    """
+    vector = vector_from_pb(proto_data.translation_vector)
+    result = Translation.from_vector(vector)
+    result.name = proto_data.name
+    result._guid = proto_data.guid
+    return result
+
+
+# =============================================================================
+# Rotation
+# =============================================================================
```

Comment:
I like these section headers, could you please add them to all other sections of this module?